stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/MessageComposerViewModel.kt

```diff
@@ -14,14 +14,16 @@ import io.getstream.chat.android.client.models.Attachment
 import io.getstream.chat.android.client.models.Message
 import io.getstream.chat.android.core.internal.InternalStreamChatApi
 import io.getstream.chat.android.offline.ChatDomain
+import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.launch
 
 /**
  * ViewModel responsible for handling the composing and sending of messages.
  */
-@InternalStreamChatApi
-public open class MessageComposerViewModel(
+
+public open class MessageComposerViewModel @InternalStreamChatApi constructor(
```

Comment:
We'll probably need to provide access to the constructor but remove the `open` modifier. Devs will need to create view model factory to pass data in the constructor (if they aren't using screen components).

Reply:
And we won't inherit from the view model anywhere hopefully 🤞🏼